cassovary-core/src/main/scala/com/twitter/cassovary/util/io/NodeStream.scala

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2014 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.twitter.cassovary.util.io
+
+import com.twitter.cassovary.graph.NodeIdEdgesMaxId
+import java.io.Closeable
+
+abstract class NodeStream extends Iterable[NodeIdEdgesMaxId] with Closeable{
```

Comment:
why not a trait?

trait NodeStream extends Iterable[NodeIdEdgesMaxId] with Closeable